src/main/java/org/apache/mesos/hdfs/executor/AbstractNodeExecutor.java

```diff
@@ -280,6 +289,64 @@ public void shutdown(ExecutorDriver d) {
     log.info("Executor asked to shutdown");
   }
 
+  protected void runHealthChecks(ExecutorDriver driver, Task task) {
+    log.info("Performing health check for task: " + task.taskInfo.getTaskId().getValue());
+    boolean taskHealthy = false;
+    String taskIdStr = task.taskInfo.getTaskId().getValue();
+    int healthCheckPort = -1;
+
+    if (taskIdStr.contains(HDFSConstants.DATA_NODE_ID)) {
+      healthCheckPort = HDFSConstants.DATA_NODE_PORT;
+    } else if (taskIdStr.contains(HDFSConstants.JOURNAL_NODE_ID)) {
+      healthCheckPort = HDFSConstants.JOURNAL_NODE_PORT;
+    } else if (taskIdStr.contains(HDFSConstants.ZKFC_NODE_ID)) {
+      healthCheckPort = HDFSConstants.ZKFC_NODE_PORT;
+    } else if (taskIdStr.contains(HDFSConstants.NAME_NODE_ID)) {
+      healthCheckPort = HDFSConstants.NAME_NODE_PORT;
+    } else {
+      log.error("Task unknown: " + taskIdStr);
+    }
+    if (healthCheckPort != -1) {
+      Socket socket = null;
+      try
+      {
+        String hostAddress = InetAddress.getLocalHost().getHostAddress();
+        socket = new Socket();
+        socket.bind(new InetSocketAddress(hostAddress, healthCheckPort));
+      } catch (IOException | SecurityException | IllegalArgumentException e) {
+        taskHealthy = true;
+        log.info("Could not bind to port " + healthCheckPort + ", socket is in use as expected.");
+      } finally {
+        if (socket != null)
+          try {
+            socket.close();
+          } catch (IOException e) {
+            log.error("Error closing socket in health check: ", e);
+          }
+      }
+    }
+    if (!taskHealthy) {
+      log.fatal("Node health check failed for task: " + taskIdStr);
+      killTask(driver, task.taskInfo.getTaskId());
+      System.exit(2);
```

Comment:
So we exit the entire executor if any of its tasks are unhealthy? Seems like we'd be better off just sending a TASK_RUNNING(unhealthy) to the scheduler, and let it decide whether to kill the unhealthy task, wait, or kill the entire executor.